app/services/katello/pulp3/generic_content_unit.rb

```diff
@@ -0,0 +1,29 @@
+module Katello
+  module Pulp3
+    class GenericContentUnit < PulpContentUnit
+      include LazyAccessor
+      CONTENT_TYPE = "generic_content_unit".freeze
+
+      def self.fetch_content_list(page_opts, repository_type)
+        content_unit_list page_opts, repository_type
+      end
+
+      def self.content_unit_list(page_opts, repository_type)
+        self.content_api(repository_type).list page_opts
+      end
+
+      def self.content_api(repository_type)
+        repository_type.content_types.first.pulp3_api.new(repository_type.pulp3_api_class.new(SmartProxy.pulp_primary!, repository_type).api_client)
```

Comment:
ah i see the .first here again.  If we want to proceed with this, maybe lets open up an issue to fix this and put it on the board.

Reply:
@jlsherrill could this be used to get the content_type from within the function `::Katello::RepositoryTypeManager.find(repository.content_type).default_managed_content_type.content_type` (assuming repository is passed instead of repository_type)? I'm not sure if this works if there's more than one content type.

Reply:
Nevermind. That works for "default" :)